### ISSUE TITLE
[intro-neural-networks/gradient-descent] Not (batch) Gradiane descent

### ISSUE THREAD

**Issue by kuzand** [udacity/deep-learning-v2-pytorch#98]
In the GradientDescent.ipynb the weights are updated after each training data example, whereas they should be updated at the end of each epoch, i.e. after processing all the training data set (batch gradient descent algorithm). So the given code actually implements the stochastic gradient descent.
This is inconsistent with the StudentsAddmissions.ipynb where the weights are updated at the end of each epoch (as they should be if we are using GD).

**Comment by abhiojha8** [comment]
We have updated the notebook.
